model/common/src/icon4py/model/common/interpolation/interpolation_fields.py

```diff
@@ -375,28 +376,36 @@ def weighting_factors(
         / ((y[2] - y[0]) - (x[2] - x[0]) * (y[1] - y[0]) / (x[1] - x[0]))
         * (1.0 - wgt_loc)
         * (-y[0] + x[0] * (y[1] - y[0]) / (x[1] - x[0])),
+        wgt[2],
+    )
+    wgt[1] = np.where(
+        mask, (-(1.0 - wgt_loc) * x[0] - wgt[2] * (x[2] - x[0])) / (x[1] - x[0]), wgt[1]
+    )
+    wgt[0] = np.where(
+        mask, 1.0 - wgt[1] - wgt[2] if wgt_loc == 0.0 else 1.0 - wgt_loc - wgt[1] - wgt[2], wgt[0]
```

Comment:
Isn't this the same as the following:
```suggestion
        mask, 1.0 - wgt_loc - wgt[1] - wgt[2], wgt[0], wgt[0]
```
Meaning, you don't need the extra if clause: since you would subtract 0.0 in the else clause anyway. And like this it becomes quite hard to read.